pkg/odo/cli/debug/portforward.go

```diff
@@ -63,9 +67,36 @@ func (o *PortForwardOptions) Complete(name string, cmd *cobra.Command, args []st
 
 	o.Context = genericclioptions.NewContext(cmd)
 	cfg, err := config.NewLocalConfigInfo(o.contextDir)
+	if err != nil {
+		return err
+	}
 	o.localConfigInfo = cfg
 
 	remotePort := cfg.GetDebugPort()
+
+	// try to listen on the given local port and check if the port is free or not
+	addressLook := "localhost:" + strconv.Itoa(o.localPort)
+	listener, err := net.Listen("tcp", addressLook)
+	if err != nil {
+		// if the local-port flag is set by the user, return the error and stop execution
+		flag := cmd.Flags().Lookup("local-port")
```

Comment:
Can you export `o.localPort` and use it here? (`o.LocalPort`)

Reply:
But the problem is that we set a default value for this and also we need to check if the default was set by the user via the flag or not.